internal/controller/clusterpolicy_controller.go

```diff
@@ -54,8 +54,9 @@ func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	_ = log.FromContext(ctx)
 	_ = r.Log.WithValues("clusterpolicy", req.NamespacedName)
 
-	clusterPolicy, err := r.GetClusterPolicy(ctx, req.NamespacedName.Name)
-	if err != nil {
+	var clusterPolicy kyvernov1.ClusterPolicy
+
+	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
 		// Error fetching the report
 
 		// Check if the ClusterPolicy was deleted
```

Comment:
Looks good. Here, if the policy was deleted, I would also check and remove it from the map.